tgui/packages/tgui/interfaces/WhitelistPanel.jsx

```diff
@@ -0,0 +1,272 @@
+            <Flex.Item bold width="20rem" shrink="0" mr="1rem">
+              {record.ckey}
+            </Flex.Item>
+            <Flex.Item width="40rem" ml="1rem" shrink="0" textAlign="center">
+              {record.status}
+            </Flex.Item>
+          </Flex>
+        );
+      })}
+    </Section>
+  );
+};
+
+const StatusUpdate = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    co_flags,
+    syn_flags,
+    yaut_flags,
+    misc_flags,
+    viewed_player,
+    user_rights,
+    target_rights,
+    new_rights,
+  } = data;
+  return (
+    <Section fill>
+      <Flex align="center">
+        <Button
+          icon="arrow-left"
+          px="2rem"
+          textAlign="center"
+          tooltip="Go back"
+          onClick={() => act('go_back')}
+        />
+      </Flex>
+      <h1 align="center">Whitelists for: {viewed_player.ckey}</h1>
+      <Section title="Commanding Officer">
+        <Stack align="right" grow={1}>
+          {co_flags.map((bit, i) => {
+            const isWhitelisted = target_rights && target_rights & bit.bitflag;
+            return (
+              <Button
+                key={i}
+                width="100%"
+                height="100%"
+                color={isWhitelisted ? 'purple' : 'blue'}
+                content={bit.name}
+              />
+            );
+          })}
+        </Stack>
+        <Stack align="right" grow={1}>
+          {co_flags.map((bit, i) => {
+            const new_state = new_rights && new_rights & bit.bitflag;
+            const editable = user_rights && bit.permission & user_rights;
+            return (
+              <Button.Checkbox
+                key={i}
+                width="100%"
+                height="100%"
+                checked={new_state}
+                color={new_state ? 'good' : 'bad'}
+                content={bit.name}
+                disabled={!editable}
+                onClick={() =>
+                  act('update_number', {
+                    'wl_flag': !new_state
+                      ? new_rights | bit.bitflag
+                      : new_rights & ~bit.bitflag,
+                  })
+                }
+              />
+            );
+          })}
+        </Stack>
+      </Section>
+      <Section title="Synthetic">
+        <Stack align="right" grow={1}>
+          {syn_flags.map((bit, i) => {
+            const isWhitelisted = target_rights && target_rights & bit.bitflag;
+            return (
+              <Button
+                key={i}
+                width="100%"
+                height="100%"
+                color={isWhitelisted ? 'purple' : 'blue'}
+                content={bit.name}
+              />
+            );
+          })}
+        </Stack>
+        <Stack align="right" grow={1}>
+          {syn_flags.map((bit, i) => {
+            const new_state = new_rights && new_rights & bit.bitflag;
+            const editable = user_rights && bit.permission & user_rights;
```

Comment:
cant do js only checks because all uis can be manipulated, these checks need to be replicated in ui_act

Reply:
All changes logged, and only four people outside of management will have access to the panel at all. I'm willing to take that risk, as if I can't trust the very limited amount of people who can access this panel to not try and abuse it they'll have a very bad day.